modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/SidebarPanelMetricsView.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React, {useEffect, useRef} from 'react';
+
+import Sidebar from './Sidebar';
+
+const SidebarPanelMetricsView = ({html}) => {
+	const elementRef = useRef();
+
+	useEffect(() => {
+		const fragment = document.createRange().createContextualFragment(html);
+
+		elementRef.current.innerHTML = '';
+
+		elementRef.current.appendChild(fragment);
+	}, [html]);
```

Comment:
This is ok, but strictly speaking you are sneaking around behind React's back when you manipulate the DOM like this. The previously "blessed" way to be sure that React would leave the DOM to you was to implement a class component that had `shouldComponentUpdate()` hard-coded to always return `false`. [The docs](https://reactjs.org/docs/react-component.html#shouldcomponentupdate) say that there is no guarantee that that will always work (ie. it might be taken as a performance hint to avoid re-rendering, but React still might render again). Having said that, the trick is used in so many places now across the internet that I think it is extremely unlikely that they would change the behavior without a loooong warning and deprecation cycle.

Reply:
Thx for the explanation :D